modules/gateway-task/variables.tf

```diff
@@ -0,0 +1,246 @@
+variable "family" {
+  description = "Task definition family (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#family). This is used by default as the Consul service name if `consul_service_name` is not provided."
+  type        = string
+}
+
+variable "consul_service_name" {
+  description = "The name the gateway service will be registered as in Consul. Defaults to the Task family name. Always suffixed with the gateway kind."
+  type        = string
+  default     = ""
+}
+
+variable "consul_service_tags" {
+  description = "A list of tags included in the Consul service registration."
+  type        = list(string)
+  default     = []
+}
+
+variable "consul_service_meta" {
+  description = "A map of metadata that will be used for the Consul service registration"
+  type        = map(string)
+  default     = {}
+}
+
+variable "consul_namespace" {
+  description = "The Consul namespace to use to register this gateway [Consul Enterprise]."
+  type        = string
+  default     = ""
+
+  validation {
+    error_message = "Gateway namespace must be 'default' or the empty string."
+    condition     = var.consul_namespace == "" || var.consul_namespace == "default"
+  }
+
+}
+
+variable "consul_partition" {
+  description = "The Consul admin partition to use to register this gateway [Consul Enterprise]."
+  type        = string
+  default     = ""
+}
+
+variable "requires_compatibilities" {
+  description = "Set of launch types required by the task."
+  type        = list(string)
+  default     = ["EC2", "FARGATE"]
+}
+
+variable "cpu" {
+  description = "Number of cpu units used by the task."
+  type        = number
+  default     = 256
+}
+
+variable "memory" {
+  description = "Amount (in MiB) of memory used by the task."
+  type        = number
+  default     = 512
+}
+
+variable "task_role" {
```

Comment:
Could we remove the task_role and execution_role variables here? They seem unlikely to be needed for gateway tasks since there is no user app running in the task.

Reply:
I think so.. It would simplify the `gateway-task` module if we alway use the generated task and execution roles.